ttnn/cpp/ttnn/operations/pool/rotate/device/rotate_nearest_program_factory.cpp

```diff
@@ -0,0 +1,245 @@
+    } else {
+        center_x = (static_cast<float>(input_width) - 1.0f) / 2.0f;
+        center_y = (static_cast<float>(input_height) - 1.0f) / 2.0f;
+    }
+
+    // Fill value as bfloat16
+    const uint16_t fill_value_bf16 = nearest_float_to_bfloat16(operation_attributes.fill);
+
+    // Work distribution - Total work units = N * H * W (one output pixel per work unit)
+    const uint32_t total_output_sticks = input_batch * input_height * input_width;
+
+    // Calculate cores needed
+    const auto compute_grid_size = device->compute_with_storage_grid_size();
+    auto [num_cores, all_cores, core_group_1, core_group_2, num_sticks_per_core_group_1, num_sticks_per_core_group_2] =
+        tt::tt_metal::split_work_to_cores(compute_grid_size, total_output_sticks);
+
+    const uint32_t num_cores_y = compute_grid_size.y;
+
+    // Get logical cores for setting runtime args
+    std::vector<CoreCoord> logical_cores = corerange_to_cores(all_cores, num_cores, true);
+
+    // Calculate stick sizes (aligned to 32 bytes)
+    const uint32_t element_size = input_tensor.element_size();
+    const uint32_t input_stick_nbytes = input_channels * element_size;
+    const uint32_t output_stick_nbytes = input_channels * element_size;
+
+    // CB_0: Output CB for communication between reader and writer
+    const uint32_t output_cb_page_size = output_stick_nbytes;
+    const auto [output_cb_index, output_cb_handle] = tt::tt_metal::create_cb(
+        tt::CBIndex::c_0, program, all_cores, output_cb_page_size, NEAREST_BUFFERING_FACTOR, output_cb_data_format);
+
+    // Reader compile-time arguments (RISCV_0)
+    std::vector<uint32_t> reader_compile_time_args = {
+        output_cb_index,     // ct_arg[0]: output_cb_index
+        input_stick_nbytes,  // ct_arg[1]: input_stick_nbytes
+        input_batch,         // ct_arg[2]: input_batch
+        input_height,        // ct_arg[3]: input_height
+        input_width,         // ct_arg[4]: input_width
+        input_channels,      // ct_arg[5]: input_channels
+    };
+
+    // Append tensor accessor args for input tensor
+    tt::tt_metal::TensorAccessorArgs(*input_tensor.buffer()).append_to(reader_compile_time_args);
+
+    // Writer compile-time arguments (RISCV_1)
+    std::vector<uint32_t> writer_compile_time_args = {
+        output_cb_index,      // ct_arg[0]: output_cb_index
+        output_stick_nbytes,  // ct_arg[1]: output_stick_nbytes
+    };
+
+    // Append tensor accessor args for output tensor
+    tt::tt_metal::TensorAccessorArgs(*output_tensor.buffer()).append_to(writer_compile_time_args);
+
+    // Create reader kernel (RISCV_0)
+    tt::tt_metal::KernelHandle reader_kernel_id = tt::tt_metal::CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/pool/rotate/device/kernels/dataflow/"
+        "reader_rotate_nearest_interleaved.cpp",
+        all_cores,
+        tt::tt_metal::DataMovementConfig{
+            .processor = tt::tt_metal::DataMovementProcessor::RISCV_0,
```

Comment:
don't hardcode risc cores /nocs use helpers to get NOCs prefered for dram read/write